SessionMessagingKit/Sending & Receiving/MessageReceiver.swift

```diff
@@ -145,7 +145,11 @@ public enum MessageReceiver {
             message.sender = sender
             message.recipient = userPublicKey
             message.sentTimestamp = envelope.timestamp
-            message.receivedTimestamp = NSDate.millisecondTimestamp()
+            if (isOpenGroupMessage) {
+                message.receivedTimestamp = envelope.serverTimestamp
```

Comment:
`envelope.serverTimestamp` is actually the * sent * timestamp of the message. Storing it in `message.receivedTimestamp` works but it's confusing.